jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -460,24 +465,10 @@ void java_bytecode_convert_methodt::convert(
     if(!is_parameter(v))
       continue;
 
-    // Construct a fully qualified name for the parameter v,
-    // e.g. my.package.ClassName.myMethodName:(II)I::anIntParam, and then a
-    // symbol_exprt with the parameter and its type
-    typet t;
-    if(v.signature.has_value())
-    {
-      t=java_type_from_string_with_exception(
-        v.descriptor,
-        v.signature,
-        id2string(class_symbol.name));
-    }
-    else
-      t=java_type_from_string(v.descriptor);
-
     std::ostringstream id_oss;
     id_oss << method_id << "::" << v.name;
     irep_idt identifier(id_oss.str());
-    symbol_exprt result(identifier, t);
+    symbol_exprt result = symbol_exprt::typeless(identifier);
```

Comment:
:question: This change looks a little suspicious? Previously the type of the symbol was derived from the signature or descriptor, now is a typeless identifier?

Reply:
The type was sometimes `nil_typet()` (because trying to derive the type failed) and nobody noticed, because really the type of this is never used.

Reply:
In that case could I suggest using an `irep_idt` instead of a `symbol_exprt`? Otherwise someone will surely assume that `variables[n].type()` is a sensible thing to look at.

Reply:
That would necessitate some refactoring of this code as the expression is also used to communicate `ID_C_base_name`.